server/utils/mpesa_utils.py

```
@@ -0,0 +1,296 @@
+
+            # Update status based on result code
+            if result_code == 0:
+                transaction.status = "completed"
+                if transaction_code:
+                    transaction.transaction_code = transaction_code
+            else:
+                transaction.status = "failed"
+
+            db.session.commit()
+
+            return transaction
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)